adapter/events.go

```diff
@@ -22,15 +22,26 @@ import (
 
 // Info is used by events to wrap user and provider info.
 type Info struct {
-	User     *UserInfo
 	Provider *ProviderInfo
 }
 
+type EventType string
```

Comment:
Turning `EventType` into a type is the right move, but the name is mildly ambiguous (a type describing an event, or a type of event). Can you add a (very) [short clarifying comment](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments)?